File: extract-pdf.js
const fs = require('fs');
const pdf = require('pdf-parse');

const dataBuffer = fs.readFileSync('C:\\Users\\devgf\\Downloads\\7. Dev_Portfolio_Knowledge_Base_ FAQ_RAG.pdf');

pdf(dataBuffer).then(function (data) {
    fs.writeFileSync('knowledge.txt', data.text);
    console.log('Done rendering pdf');
}).catch(err => console.error(err));

File: extract-pdf.py
from pypdf import PdfReader

reader = PdfReader(r'C:\Users\devgf\Downloads\7. Dev_Portfolio_Knowledge_Base_ FAQ_RAG.pdf')
text = ""
for page in reader.pages:
    text += page.extract_text() + "\n"

with open("knowledge.txt", "w", encoding="utf-8") as f:
    f.write(text)
print("done")
